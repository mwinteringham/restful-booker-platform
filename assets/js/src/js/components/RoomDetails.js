import React, { useEffect, useState } from 'react';
import BookingListings from './BookingListings.js';
import { useParams } from 'react-router-dom';
import { API } from '../libs/Api.js';

const RoomDetails = ({}) => {

    let { id } = useParams();
    const [edit, toggleEdit] = useState(false);
    const [room, setRoom] = useState({
        accessible : false,
        description : "",
        featuresObject : {
            WiFi : false,
            TV : false,
            Radio : false,
            Refreshments : false,
            Safe : false,
            Views : false
        },
        features : []
    });
    const [errors, setErrors] = useState([]);

    useEffect(() => {
        API.getRoomById(id, room, setRoom);
    }, [])

    const toggleAndRestEdit = (toggle) => {
        setErrors({});
        toggleEdit(toggle);
    }

    const doEdit = () => {
        room.features = Object.keys(room.featuresObject).filter(key => room.featuresObject[key]);

        API.putRoom(id, room, resetForm, fetchRoomDetails, setErrors);
    }

    const resetForm = () => {
        toggleEdit(false);
        setRoom({
            accessible : false,
            description : "",
            featuresObject : {
                WiFi : false,
                TV : false,
                Radio : false,
                Refreshments : false,
                Safe : false,
                Views : false
            },
            features : []
        });
        setErrors([]);
    }

    const fetchRoomDetails = () => {
        API.getRoomById(id, room, setRoom)
    }

    const updateState = (event) => {
        if(event.name === 'featureCheck'){
            room.featuresObject[event.value] = event.checked;
        } else {
            room[event.id] = event.value;
        }

        setRoom(prevState => ({
            ...prevState
        }));
    }

    let roomSummary = null;
    let errorMessages = '';
    
    if(errors.length > 0){
        errorMessages = <div className="alert alert-danger" style={{marginTop : 15 + "px"}}>
            {errors.map((value) => {
                return <p key={value}>{value}</p>
            })}
        </div>
    }

    if(edit == true){
        roomSummary =  <div className="room-details">
                            <div className="form-row">
                                <div className="col-sm-9">
                                    <h2>Room: </h2>
                                    <input type="text" defaultValue={room.roomName} id="roomName" onChange={e => updateState(e.target)} />
                                </div>
                                <div className="col-sm-3">
                                    <button onClick={() => toggleAndRestEdit(false)} type="button" id="cancelEdit" className="btn btn-outline-danger float-right">Cancel</button>
                                    <button onClick={doEdit} type="button" id="update" className="btn btn-outline-primary float-right" style={{marginRight: "10px"}}>Update</button>
                                </div>
                            </div>
                            <div className="form-row">
                                <div className="col-sm-6">
                                    <label className="editLabel" htmlFor="type">Type: </label>
                                    <select className="form-control" id="type" value={room.type} onChange={e => updateState(e.target)}>
                                        <option value="Single">Single</option>
                                        <option value="Twin">Twin</option>
                                        <option value="Double">Double</option>
                                        <option value="Family">Family</option>
                                        <option value="Suite">Suite</option>
                                    </select>
                                    <label className="editLabel" htmlFor="type">Accessible: </label>
                                    <select className="form-control" id="accessible" value={room.accessible} onChange={e => updateState(e.target)}>
                                        <option value="false">false</option>
                                        <option value="true">true</option>
                                    </select>
                                    <label className="editLabel" htmlFor="roomPrice">Room price: </label>
                                    <input className="form-control" type="text" defaultValue={room.roomPrice} id="roomPrice" onChange={e => updateState(e.target)} />
                                </div>
                                <div className="col-sm-6">
                                    <label className="editLabel" htmlFor="description">Description: </label>
                                    <textarea className="form-control" aria-label="Description" defaultValue={room.description} id="description" rows="5" onChange={e => updateState(e.target)}></textarea>
                                </div>
                            </div>
                            <div className="form-row">
                                <div className="col-sm-6">
                                    <label className="editLabel">Room features: </label>
                                    <div className="row">
                                        <div className="col-4">
                                            <div className="form-check form-check-inline">
                                                <input className="form-check-input" type="checkbox" name="featureCheck" id="wifiCheckbox" value="WiFi" checked={room.featuresObject.WiFi} onChange={e => updateState(e.target)} />
                                                <label className="form-check-label" htmlFor="wifiCheckbox">WiFi</label>
                                            </div>
                                        </div>
                                        <div className="col-4">
                                            <div className="form-check form-check-inline">
                                                <input className="form-check-input" type="checkbox" name="featureCheck" id="tvCheckbox" value="TV" checked={room.featuresObject.TV} onChange={e => updateState(e.target)} />
                                                <label className="form-check-label" htmlFor="tvCheckbox">TV</label>
                                            </div>
                                        </div>
                                        <div className="col-4">
                                            <div className="form-check form-check-inline">
                                                <input className="form-check-input" type="checkbox" name="featureCheck" id="radioCheckbox" value="Radio" checked={room.featuresObject.Radio} onChange={e => updateState(e.target)} />
                                                <label className="form-check-label" htmlFor="radioCheckbox">Radio</label>
                                            </div>
                                        </div>
                                    </div>
                                    <div className="row">
                                        <div className="col-4">
                                            <div className="form-check form-check-inline">
                                                <input className="form-check-input" type="checkbox" name="featureCheck" id="refreshCheckbox" value="Refreshments" checked={room.featuresObject.Refreshments} onChange={e => updateState(e.target)} />
                                                <label className="form-check-label" htmlFor="refreshCheckbox">Refreshments</label>
                                            </div>
                                        </div>
                                        <div className="col-4">
                                            <div className="form-check form-check-inline">
                                                <input className="form-check-input" type="checkbox" name="featureCheck" id="safeCheckbox" value="Safe" checked={room.featuresObject.Safe} onChange={e => updateState(e.target)} />
                                                <label className="form-check-label" htmlFor="safeCheckbox">Safe</label>
                                            </div>
                                        </div>
                                        <div className="col-4">
                                            <div className="form-check form-check-inline">
                                                <input className="form-check-input" type="checkbox" name="featureCheck" id="viewsCheckbox" value="Views" checked={room.featuresObject.Views} onChange={e => updateState(e.target)} />
                                                <label className="form-check-label" htmlFor="viewsCheckbox">Views</label>
                                            </div>
                                        </div>
                                    </div>
                                </div>
                                <div className="col-sm-6">
                                    <label className="editLabel" htmlFor="image">Image: </label>
                                    <input type="text" className="form-control" defaultValue={room.image} id="image" onChange={e => updateState(e.target)} />
                                </div>
                            </div>
                            {errorMessages}
                        </div>
    } else {
        roomSummary = <div className="room-details">
                        <div className="row">
                            <div className="col-sm-10">
                                <h2>Room: {room.roomName}</h2>
                            </div>
                            <div className="col-sm-2">
                                <button onClick={() => toggleAndRestEdit(true)} type="button" className="btn btn-outline-primary float-right" style={{marginRight: "10px"}}>Edit</button>
                            </div>
                        </div>
                        <div className="row">
                            <div className="col-sm-6">
                                <p>Type: <span>{room.type}</span></p>
                            </div>
                            <div className="col-sm-6">
                                <p>Description: <span>{room.description.length >= 50 &&
                                    room.description.substring(0,50) + "..."
                                }
                                {room.description.length < 50 &&
                                    room.description
                                }</span></p>
                            </div>
                        </div>
                        <div className="row">
                            <div className="col-sm-6">
                                <p>Accessible: <span>{room.accessible.toString()}</span></p>
                                <p>Features: <span>
                                    {room.features.length > 0 &&
                                        room.features.map((value, index) => {
                                            if(index + 1 === room.features.length){
                                                return value;
                                            } else {
                                                return value + ", ";
                                            }
                                        })
                                    }
                                    {room.features.length === 0 && 
                                        <span style={{color : "grey"}}>No features added to the room</span>
                                    }
                                </span></p>
                                <p>Room price: <span>{room.roomPrice}</span></p>
                            </div>
                            <div className="col-sm-6">
                                <p>Image:</p>
                                <img src={room.image} alt={"Room: " + room.roomName + " preview image"} />
                            </div>
                        </div>
                    </div>
    }

    return(
        <div>
            {roomSummary}
            <div className="row">
                <div className="col-sm-2 rowHeader"><p>First name</p></div>
                <div className="col-sm-2 rowHeader"><p>Last name</p></div>
                <div className="col-sm-1 rowHeader"><p>Price</p></div>
                <div className="col-sm-2 rowHeader"><p>Deposit paid?</p></div>
                <div className="col-sm-2 rowHeader"><p>Check in</p></div>
                <div className="col-sm-2 rowHeader"><p>Check out</p></div>
                <div className="col-sm-1"></div>
            </div>
            <BookingListings roomid={id} roomPrice={room.roomPrice} />
        </div>
    )

}

export default RoomDetails;
