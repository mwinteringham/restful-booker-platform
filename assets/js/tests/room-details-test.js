import React from 'react';
import RoomDetails from '../src/js/components/RoomDetails.js';
import { Routes, Route, Router } from 'react-router-dom';
import {createMemoryHistory} from 'history'
import nock from 'nock';

import {
    render, waitFor, fireEvent
  } from '@testing-library/react'

const roomObject = {
    roomid: 1,
    roomName: "101",
    type: "Single",
    accessible: true,
    image: "https://www.mwtestconsultancy.co.uk/img/testim/room2.jpg",
    description: "Aenean porttitor mauris sit amet lacinia molestie. In posuere accumsan aliquet. Maecenas sit amet nisl massa. Interdum et malesuada fames ac ante.",
    features: [
        "TV, WiFi, Safe"
    ],
    roomPrice: 100
}

const history = createMemoryHistory()
history.push('/admin/room/1')

nock('http://localhost')
    .persist()
    .get('/booking/?roomid=1')
    .reply(200, {
        bookings : []
    })

test('Room details component renders', async () => {
    const roomMock = nock('http://localhost')
                        .get('/room/1')
                        .reply(200, roomObject);

    const {asFragment, getByText} = render(
        <Router location={history.location} navigator={history}>
            <Routes>
                <Route path="/admin/room/:id" element={<RoomDetails />} />
            </Routes>
        </Router>
    )

    await waitFor(() => expect(roomMock.isDone()).toBeTruthy())
    await waitFor(() => expect(getByText(/101/)).not.toBeNull());
    
    expect(asFragment()).toMatchSnapshot();
});

test('Room details switches into edit mode', async () => {
    const roomMock = nock('http://localhost')
                        .get('/room/1')
                        .reply(200, roomObject);

    const {asFragment, getByText} = render(
        <Router location={history.location} navigator={history}>
            <Routes>
                <Route path="/admin/room/:id" element={<RoomDetails />} />
            </Routes>
        </Router>
    )

    await waitFor(() => {expect(roomMock.isDone()).toBeTruthy()})
    
    fireEvent.click(getByText(/Edit/));

    await getByText(/Update/)

    expect(asFragment()).toMatchSnapshot();
});

test('Room details can be switched out of edit mode', async () => {
    const roomMock = nock('http://localhost')
                        .get('/room/1')
                        .reply(200, roomObject);

    const {asFragment, getByText} = render(
        <Router location={history.location} navigator={history}>
            <Routes>
                <Route path="/admin/room/:id" element={<RoomDetails />} />
            </Routes>
        </Router>
    )

    await waitFor(() => {expect(roomMock.isDone()).toBeTruthy()})
    
    fireEvent.click(getByText(/Edit/));
    fireEvent.click(getByText(/Cancel/));

    expect(asFragment()).toMatchSnapshot();
});

test('Room details can render validation errors', async () => {
    const roomMock = nock('http://localhost')
                        .get('/room/1')
                        .reply(200, roomObject);

    const putMock = nock('http://localhost')
                        .put('/room/1')
                        .reply(400, {
                            "errorCode": 400,
                            "error": "BAD_REQUEST",
                            "errorMessage": "Validation failed for argument [0] in public org.springframework.http.ResponseEntity<com.automationintesting.model.db.Room> com.automationintesting.api.RoomController.updateRoom(com.automationintesting.model.db.Room,int,java.lang.String) throws java.sql.SQLException: [Field error in object 'room' on field 'type': rejected value [single]; codes [Pattern.room.type,Pattern.type,Pattern.java.lang.String,Pattern]; arguments [org.springframework.context.support.DefaultMessageSourceResolvable: codes [room.type,type]; arguments []; default message [type],[Ljavax.validation.constraints.Pattern$Flag;@21a98ac5,Single|Double|Twin|Family|Suite]; default message [Type can only contain the room options Single, Double, Twin, Family or Suite]] ",
                            "fieldErrors": ["Type can only contain the room options Single, Double, Twin, Family or Suite"]
                        });

    const {asFragment, getByText} = render(
        <Router location={history.location} navigator={history}>
            <Routes>
                <Route path="/admin/room/:id" element={<RoomDetails />} />
            </Routes>
        </Router>
    )

    await waitFor(() => {expect(roomMock.isDone()).toBeTruthy()})

    fireEvent.click(getByText(/Edit/));
    fireEvent.click(getByText(/Update/));

    await waitFor(() => {expect(putMock.isDone()).toBeTruthy()})

    expect(asFragment()).toMatchSnapshot();
});

test('Room details can be submitted', async () => {
    const roomMock = nock('http://localhost')
                        .persist()
                        .get('/room/1')
                        .reply(200, roomObject);

    let putMock = nock('http://localhost')
                    .put('/room/1', {
                        "roomid": 1,
                        "roomName": "101",
                        "type": "Single",
                        "accessible": true,
                        "image": "https://www.mwtestconsultancy.co.uk/img/testim/room2.jpg",
                        "description": "Aenean porttitor mauris sit amet lacinia molestie. In posuere accumsan aliquet. Maecenas sit amet nisl massa. Interdum et malesuada fames ac ante.",
                        "features": ["TV, WiFi, Safe"],
                        "roomPrice": 100,
                        "featuresObject": {
                            "WiFi": false,
                            "TV": false,
                            "Radio": false,
                            "Refreshments": false,
                            "Safe": false,
                            "Views": false,
                            "TV, WiFi, Safe": true
                        }
                    })
                    .reply(202);

    const {getByText} = render(
        <Router location={history.location} navigator={history}>
            <Routes>
                <Route path="/admin/room/:id" element={<RoomDetails />} />
            </Routes>
        </Router>
    )

    await waitFor(() => {expect(roomMock.isDone()).toBeTruthy()})

    fireEvent.click(getByText(/Edit/));
    fireEvent.click(getByText(/Update/));

    await waitFor(() => {expect(putMock.isDone()).toBeTruthy()})
});
