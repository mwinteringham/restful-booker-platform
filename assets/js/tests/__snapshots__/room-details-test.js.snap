// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Room details can be switched out of edit mode 1`] = `
<DocumentFragment>
  <div>
    <div
      class="room-details"
    >
      <div
        class="row"
      >
        <div
          class="col-sm-10"
        >
          <h2>
            Room: 101
          </h2>
        </div>
        <div
          class="col-sm-2"
        >
          <button
            class="btn btn-outline-primary float-right"
            style="margin-right: 10px;"
            type="button"
          >
            Edit
          </button>
        </div>
      </div>
      <div
        class="row"
      >
        <div
          class="col-sm-6"
        >
          <p>
            Type: 
            <span>
              Single
            </span>
          </p>
        </div>
        <div
          class="col-sm-6"
        >
          <p>
            Description: 
            <span>
              Aenean porttitor mauris sit amet lacinia molestie....
            </span>
          </p>
        </div>
      </div>
      <div
        class="row"
      >
        <div
          class="col-sm-6"
        >
          <p>
            Accessible: 
            <span>
              true
            </span>
          </p>
          <p>
            Features: 
            <span>
              TV, WiFi, Safe
            </span>
          </p>
          <p>
            Room price: 
            <span>
              100
            </span>
          </p>
        </div>
        <div
          class="col-sm-6"
        >
          <p>
            Image:
          </p>
          <img
            alt="Room: 101 preview image"
            src="https://www.mwtestconsultancy.co.uk/img/testim/room2.jpg"
          />
        </div>
      </div>
    </div>
    <div
      class="row"
    >
      <div
        class="col-sm-2 rowHeader"
      >
        <p>
          First name
        </p>
      </div>
      <div
        class="col-sm-2 rowHeader"
      >
        <p>
          Last name
        </p>
      </div>
      <div
        class="col-sm-1 rowHeader"
      >
        <p>
          Price
        </p>
      </div>
      <div
        class="col-sm-2 rowHeader"
      >
        <p>
          Deposit paid?
        </p>
      </div>
      <div
        class="col-sm-2 rowHeader"
      >
        <p>
          Check in
        </p>
      </div>
      <div
        class="col-sm-2 rowHeader"
      >
        <p>
          Check out
        </p>
      </div>
      <div
        class="col-sm-1"
      />
    </div>
    <div />
  </div>
</DocumentFragment>
`;

exports[`Room details can render validation errors 1`] = `
<DocumentFragment>
  <div>
    <div
      class="room-details"
    >
      <div
        class="form-row"
      >
        <div
          class="col-sm-9"
        >
          <h2>
            Room: 
          </h2>
          <input
            id="roomName"
            type="text"
            value="101"
          />
        </div>
        <div
          class="col-sm-3"
        >
          <button
            class="btn btn-outline-danger float-right"
            id="cancelEdit"
            style="margin-right: 10px;"
            type="button"
          >
            Cancel
          </button>
          <button
            class="btn btn-outline-primary float-right"
            id="update"
            style="margin-right: 10px;"
            type="button"
          >
            Update
          </button>
        </div>
      </div>
      <div
        class="form-row"
      >
        <div
          class="col-sm-6"
        >
          <label
            class="editLabel"
            for="type"
          >
            Type: 
          </label>
          <select
            class="form-control"
            id="type"
          >
            <option
              value="Single"
            >
              Single
            </option>
            <option
              value="Twin"
            >
              Twin
            </option>
            <option
              value="Double"
            >
              Double
            </option>
            <option
              value="Family"
            >
              Family
            </option>
            <option
              value="Suite"
            >
              Suite
            </option>
          </select>
          <label
            class="editLabel"
            for="type"
          >
            Accessible: 
          </label>
          <select
            class="form-control"
            id="accessible"
          >
            <option
              value="false"
            >
              false
            </option>
            <option
              value="true"
            >
              true
            </option>
          </select>
          <label
            class="editLabel"
            for="roomPrice"
          >
            Room price: 
          </label>
          <input
            class="form-control"
            id="roomPrice"
            type="text"
            value="100"
          />
        </div>
        <div
          class="col-sm-6"
        >
          <label
            class="editLabel"
            for="description"
          >
            Description: 
          </label>
          <textarea
            aria-label="Description"
            class="form-control"
            id="description"
            rows="5"
          >
            Aenean porttitor mauris sit amet lacinia molestie. In posuere accumsan aliquet. Maecenas sit amet nisl massa. Interdum et malesuada fames ac ante.
          </textarea>
        </div>
      </div>
      <div
        class="form-row"
      >
        <div
          class="col-sm-6"
        >
          <label
            class="editLabel"
          >
            Room features: 
          </label>
          <div
            class="row"
          >
            <div
              class="col-4"
            >
              <div
                class="form-check form-check-inline"
              >
                <input
                  class="form-check-input"
                  id="wifiCheckbox"
                  name="featureCheck"
                  type="checkbox"
                  value="WiFi"
                />
                <label
                  class="form-check-label"
                  for="wifiCheckbox"
                >
                  WiFi
                </label>
              </div>
            </div>
            <div
              class="col-4"
            >
              <div
                class="form-check form-check-inline"
              >
                <input
                  class="form-check-input"
                  id="tvCheckbox"
                  name="featureCheck"
                  type="checkbox"
                  value="TV"
                />
                <label
                  class="form-check-label"
                  for="tvCheckbox"
                >
                  TV
                </label>
              </div>
            </div>
            <div
              class="col-4"
            >
              <div
                class="form-check form-check-inline"
              >
                <input
                  class="form-check-input"
                  id="radioCheckbox"
                  name="featureCheck"
                  type="checkbox"
                  value="Radio"
                />
                <label
                  class="form-check-label"
                  for="radioCheckbox"
                >
                  Radio
                </label>
              </div>
            </div>
          </div>
          <div
            class="row"
          >
            <div
              class="col-4"
            >
              <div
                class="form-check form-check-inline"
              >
                <input
                  class="form-check-input"
                  id="refreshCheckbox"
                  name="featureCheck"
                  type="checkbox"
                  value="Refreshments"
                />
                <label
                  class="form-check-label"
                  for="refreshCheckbox"
                >
                  Refreshments
                </label>
              </div>
            </div>
            <div
              class="col-4"
            >
              <div
                class="form-check form-check-inline"
              >
                <input
                  class="form-check-input"
                  id="safeCheckbox"
                  name="featureCheck"
                  type="checkbox"
                  value="Safe"
                />
                <label
                  class="form-check-label"
                  for="safeCheckbox"
                >
                  Safe
                </label>
              </div>
            </div>
            <div
              class="col-4"
            >
              <div
                class="form-check form-check-inline"
              >
                <input
                  class="form-check-input"
                  id="viewsCheckbox"
                  name="featureCheck"
                  type="checkbox"
                  value="Views"
                />
                <label
                  class="form-check-label"
                  for="viewsCheckbox"
                >
                  Views
                </label>
              </div>
            </div>
          </div>
        </div>
        <div
          class="col-sm-6"
        >
          <label
            class="editLabel"
            for="image"
          >
            Image: 
          </label>
          <input
            class="form-control"
            id="image"
            type="text"
            value="https://www.mwtestconsultancy.co.uk/img/testim/room2.jpg"
          />
        </div>
      </div>
      <div
        class="alert alert-danger"
        style="margin-top: 15px;"
      >
        <p>
          Type can only contain the room options Single, Double, Twin, Family or Suite
        </p>
      </div>
    </div>
    <div
      class="row"
    >
      <div
        class="col-sm-2 rowHeader"
      >
        <p>
          First name
        </p>
      </div>
      <div
        class="col-sm-2 rowHeader"
      >
        <p>
          Last name
        </p>
      </div>
      <div
        class="col-sm-1 rowHeader"
      >
        <p>
          Price
        </p>
      </div>
      <div
        class="col-sm-2 rowHeader"
      >
        <p>
          Deposit paid?
        </p>
      </div>
      <div
        class="col-sm-2 rowHeader"
      >
        <p>
          Check in
        </p>
      </div>
      <div
        class="col-sm-2 rowHeader"
      >
        <p>
          Check out
        </p>
      </div>
      <div
        class="col-sm-1"
      />
    </div>
    <div />
  </div>
</DocumentFragment>
`;

exports[`Room details component renders 1`] = `
<DocumentFragment>
  <div>
    <div
      class="room-details"
    >
      <div
        class="row"
      >
        <div
          class="col-sm-10"
        >
          <h2>
            Room: 101
          </h2>
        </div>
        <div
          class="col-sm-2"
        >
          <button
            class="btn btn-outline-primary float-right"
            style="margin-right: 10px;"
            type="button"
          >
            Edit
          </button>
        </div>
      </div>
      <div
        class="row"
      >
        <div
          class="col-sm-6"
        >
          <p>
            Type: 
            <span>
              Single
            </span>
          </p>
        </div>
        <div
          class="col-sm-6"
        >
          <p>
            Description: 
            <span>
              Aenean porttitor mauris sit amet lacinia molestie....
            </span>
          </p>
        </div>
      </div>
      <div
        class="row"
      >
        <div
          class="col-sm-6"
        >
          <p>
            Accessible: 
            <span>
              true
            </span>
          </p>
          <p>
            Features: 
            <span>
              TV, WiFi, Safe
            </span>
          </p>
          <p>
            Room price: 
            <span>
              100
            </span>
          </p>
        </div>
        <div
          class="col-sm-6"
        >
          <p>
            Image:
          </p>
          <img
            alt="Room: 101 preview image"
            src="https://www.mwtestconsultancy.co.uk/img/testim/room2.jpg"
          />
        </div>
      </div>
    </div>
    <div
      class="row"
    >
      <div
        class="col-sm-2 rowHeader"
      >
        <p>
          First name
        </p>
      </div>
      <div
        class="col-sm-2 rowHeader"
      >
        <p>
          Last name
        </p>
      </div>
      <div
        class="col-sm-1 rowHeader"
      >
        <p>
          Price
        </p>
      </div>
      <div
        class="col-sm-2 rowHeader"
      >
        <p>
          Deposit paid?
        </p>
      </div>
      <div
        class="col-sm-2 rowHeader"
      >
        <p>
          Check in
        </p>
      </div>
      <div
        class="col-sm-2 rowHeader"
      >
        <p>
          Check out
        </p>
      </div>
      <div
        class="col-sm-1"
      />
    </div>
    <div />
  </div>
</DocumentFragment>
`;

exports[`Room details switches into edit mode 1`] = `
<DocumentFragment>
  <div>
    <div
      class="room-details"
    >
      <div
        class="form-row"
      >
        <div
          class="col-sm-9"
        >
          <h2>
            Room: 
          </h2>
          <input
            id="roomName"
            type="text"
            value="101"
          />
        </div>
        <div
          class="col-sm-3"
        >
          <button
            class="btn btn-outline-danger float-right"
            id="cancelEdit"
            style="margin-right: 10px;"
            type="button"
          >
            Cancel
          </button>
          <button
            class="btn btn-outline-primary float-right"
            id="update"
            style="margin-right: 10px;"
            type="button"
          >
            Update
          </button>
        </div>
      </div>
      <div
        class="form-row"
      >
        <div
          class="col-sm-6"
        >
          <label
            class="editLabel"
            for="type"
          >
            Type: 
          </label>
          <select
            class="form-control"
            id="type"
          >
            <option
              value="Single"
            >
              Single
            </option>
            <option
              value="Twin"
            >
              Twin
            </option>
            <option
              value="Double"
            >
              Double
            </option>
            <option
              value="Family"
            >
              Family
            </option>
            <option
              value="Suite"
            >
              Suite
            </option>
          </select>
          <label
            class="editLabel"
            for="type"
          >
            Accessible: 
          </label>
          <select
            class="form-control"
            id="accessible"
          >
            <option
              value="false"
            >
              false
            </option>
            <option
              value="true"
            >
              true
            </option>
          </select>
          <label
            class="editLabel"
            for="roomPrice"
          >
            Room price: 
          </label>
          <input
            class="form-control"
            id="roomPrice"
            type="text"
            value="100"
          />
        </div>
        <div
          class="col-sm-6"
        >
          <label
            class="editLabel"
            for="description"
          >
            Description: 
          </label>
          <textarea
            aria-label="Description"
            class="form-control"
            id="description"
            rows="5"
          >
            Aenean porttitor mauris sit amet lacinia molestie. In posuere accumsan aliquet. Maecenas sit amet nisl massa. Interdum et malesuada fames ac ante.
          </textarea>
        </div>
      </div>
      <div
        class="form-row"
      >
        <div
          class="col-sm-6"
        >
          <label
            class="editLabel"
          >
            Room features: 
          </label>
          <div
            class="row"
          >
            <div
              class="col-4"
            >
              <div
                class="form-check form-check-inline"
              >
                <input
                  class="form-check-input"
                  id="wifiCheckbox"
                  name="featureCheck"
                  type="checkbox"
                  value="WiFi"
                />
                <label
                  class="form-check-label"
                  for="wifiCheckbox"
                >
                  WiFi
                </label>
              </div>
            </div>
            <div
              class="col-4"
            >
              <div
                class="form-check form-check-inline"
              >
                <input
                  class="form-check-input"
                  id="tvCheckbox"
                  name="featureCheck"
                  type="checkbox"
                  value="TV"
                />
                <label
                  class="form-check-label"
                  for="tvCheckbox"
                >
                  TV
                </label>
              </div>
            </div>
            <div
              class="col-4"
            >
              <div
                class="form-check form-check-inline"
              >
                <input
                  class="form-check-input"
                  id="radioCheckbox"
                  name="featureCheck"
                  type="checkbox"
                  value="Radio"
                />
                <label
                  class="form-check-label"
                  for="radioCheckbox"
                >
                  Radio
                </label>
              </div>
            </div>
          </div>
          <div
            class="row"
          >
            <div
              class="col-4"
            >
              <div
                class="form-check form-check-inline"
              >
                <input
                  class="form-check-input"
                  id="refreshCheckbox"
                  name="featureCheck"
                  type="checkbox"
                  value="Refreshments"
                />
                <label
                  class="form-check-label"
                  for="refreshCheckbox"
                >
                  Refreshments
                </label>
              </div>
            </div>
            <div
              class="col-4"
            >
              <div
                class="form-check form-check-inline"
              >
                <input
                  class="form-check-input"
                  id="safeCheckbox"
                  name="featureCheck"
                  type="checkbox"
                  value="Safe"
                />
                <label
                  class="form-check-label"
                  for="safeCheckbox"
                >
                  Safe
                </label>
              </div>
            </div>
            <div
              class="col-4"
            >
              <div
                class="form-check form-check-inline"
              >
                <input
                  class="form-check-input"
                  id="viewsCheckbox"
                  name="featureCheck"
                  type="checkbox"
                  value="Views"
                />
                <label
                  class="form-check-label"
                  for="viewsCheckbox"
                >
                  Views
                </label>
              </div>
            </div>
          </div>
        </div>
        <div
          class="col-sm-6"
        >
          <label
            class="editLabel"
            for="image"
          >
            Image: 
          </label>
          <input
            class="form-control"
            id="image"
            type="text"
            value="https://www.mwtestconsultancy.co.uk/img/testim/room2.jpg"
          />
        </div>
      </div>
    </div>
    <div
      class="row"
    >
      <div
        class="col-sm-2 rowHeader"
      >
        <p>
          First name
        </p>
      </div>
      <div
        class="col-sm-2 rowHeader"
      >
        <p>
          Last name
        </p>
      </div>
      <div
        class="col-sm-1 rowHeader"
      >
        <p>
          Price
        </p>
      </div>
      <div
        class="col-sm-2 rowHeader"
      >
        <p>
          Deposit paid?
        </p>
      </div>
      <div
        class="col-sm-2 rowHeader"
      >
        <p>
          Check in
        </p>
      </div>
      <div
        class="col-sm-2 rowHeader"
      >
        <p>
          Check out
        </p>
      </div>
      <div
        class="col-sm-1"
      />
    </div>
    <div />
  </div>
</DocumentFragment>
`;
